webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -42,12 +50,58 @@ const CodeEditor = React.lazy(() => import('hew/CodeEditor'));
 
 const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }: Props = {}) => {
   const idPrefix = useId();
-  const { canModifyWorkspaceAgentUserGroup, canModifyWorkspaceCheckpointStorage } =
-    usePermissions();
+  const {
+    canModifyWorkspaceAgentUserGroup,
+    canModifyWorkspaceCheckpointStorage,
+    canSetWorkspaceNamespaceBindings,
+  } = usePermissions();
   const [form] = Form.useForm<FormInputs>();
   const useAgentUser = Form.useWatch('useAgentUser', form);
   const useAgentGroup = Form.useWatch('useAgentGroup', form);
   const useCheckpointStorage = Form.useWatch('useCheckpointStorage', form);
+  const resourceManagers = useAsync(async (canceller) => {
+    try {
+      const response = await getKubernetesResourceManagers(undefined, { signal: canceller.signal });
+      return response.names;
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Failed to fetch Resource Managers.',
+        silent: false,
+        type: ErrorType.Server,
+      });
+      return NotLoaded;
+    }
+  }, []);
+
+  const isNonK8RMError = (e: unknown): boolean => {
+    return e instanceof DetError && (e.publicMessage?.includes('operation not supported') ?? false);
+  };
```

Comment:
this function can be outside of the component and probably wanna use `===` instead of `includes` if thats the entire message?

Reply:
It's not the entire message. Just for my understanding, is there a benefit of putting it outside the component?

Reply:
its more performant

Reply:
moved it outside!